### ISSUE TITLE
DefaultNamingSubSystem.InvalidateCache was protected and has been made private in 5.1.0

### ISSUE THREAD

**Issue by Echtelion** [castleproject/Windsor#566]
Hello,
refer to #151

I'm sorry to reopen this old ticket, but the 5.0 version of windsor changed InvalidateCache to private.
It's also not possible to reimplement it in the derived method, because the fields are also made private and the property are protected read only. Ultimately, assignableHandlerListsByTypeCache is private with no protected accessor at all.

I can reimplement it with reflection, as I'm not using this method intensively, but is there a new recommended way to do this ?

Regards,

**Comment by nativenolde** [comment]
@generik0 Unfortunately, i am also affected by this change...

https://github.com/castleproject/Windsor/pull/556/files#diff-abfeffa5864abf0d2f22d3a8eb387e9742a306bb7309ad528b044b78da178b45

@Echtelion You can use that snippet as a workaround:

```
        private void InvalidateCacheUsingReflection()
        {
            const BindingFlags flags = BindingFlags.Instance | BindingFlags.NonPublic;
            var method = typeof(DefaultNamingSubSystem).GetMethod("InvalidateCache", flags);
            if (method != null)
            {
                method.Invoke(this, null);
            }
        }
```

**Comment by generik0** [comment]
@Echtelion with the latest release?

**Comment by generik0** [comment]
@Echtelion @nativenolde 

Please tell me What the exact issue is. 
It is private now, you where using in a derived cleans, and don't have access?

Is this also the issue for you @nativenolde just how it is exposed, or is there an
Issue withe the functionality of the change?

**Comment by Echtelion** [comment]
The commit is for "Add InvalidateCache to DependencyInjectionNamingSubsystem", when it actually remove the ability to be called from others derived Subsystem.

The reason I'm using this method is for being able to unregister an instance. After removing it, I have to clear the cache in order to have the container behave normally.
I now have to drop the whole container and create a new one, instead of reusing an existing container. I managed to pass existing instances from old to new containers when possible, so I have a workaround.

**Comment by generik0** [comment]
@Echtelion can you make a PR with the change you need to fix it?
Would a protected virtual help?

**Comment by Echtelion** [comment]
Yes, a protected virtual will do the job.

Sorry, I'm not used to git, still under tfs

**Comment by generik0** [comment]
Tfs has got ;-)

**Comment by Echtelion** [comment]
Thanks.

Not a problem for me, I have a workaround to wait, and in the worst case I can compile it myself.

**Comment by jonorossi** [comment]
Fixed by #569.
